File: 주영/Any/605. Can Place Flowers.js
/**
 * @param {number[]} flowerbed
 * @param {number} n
 * @return {boolean}
 */
var canPlaceFlowers = function (flowerbed, n) {
  flowerbed = [0, ...flowerbed, 0];

  for (let i = 0; i < flowerbed.length; i++) {
    if (
      flowerbed[i] === 0 &&
      flowerbed[i + 1] === 0 &&
      flowerbed[i - 1] === 0
    ) {
      n--;
      i++;
    }

    if (!n) break;
  }
  console.log('n: ', n);

  return n === 0;
};

let flowerbed = [0, 0, 1, 0, 1];

let n = 1;

canPlaceFlowers(flowerbed, n);

File: P/605_Can_Place_Flowers.java
class Solution {
    public boolean canPlaceFlowers(int[] flowerbed, int n) {
        for(int i = 0; i < flowerbed.length; ++i) {
            if(flowerbed[i] == 1) {
                i++;
            } else {
                int left =  i - 1 >= 0 ? flowerbed[i - 1] : 0;
                int right = i + 1 < flowerbed.length ? flowerbed[i + 1] : 0;
                if(left == 0 && right == 0) {
                    n--;
                    i++;
                }
            }
        }
        return n <= 0;
    }
}